### PR TITLE
Add boost cpu reference for subgraph matching

### PR DESCRIPTION
**Pull request by Laurawly**
@neoblizz @porumbes Could you help review?

### CODE REVIEW

File: gunrock/app/sm/sm_test.cuh [hunk 1]
```diff
@@ -87,6 +95,77 @@ void DisplaySolution(T *array, SizeT length) {
...
+               typename CorrespondenceMap2To1>
+      bool operator()(CorrespondenceMap1To2 f, CorrespondenceMap2To1) {
+          // Count number of matches from isomorphism map
+          (*count_ptr)++;
+          return true;
+      }
+  private:
+      const Graph1& graph1_;
+      const Graph2& graph2_;
+      size_t *count_ptr{nullptr};
```
Comment (porumbes) [inline gunrock/app/sm/sm_test.cuh]: I'm not sure I understand why the `count_ptr` variable exists? Why not increment `count_` directly?

Comment (Laurawly) [reply gunrock/app/sm/sm_test.cuh]: 'Cause boost copy the callback structure multiple times, it's impossible to accumulate count. I tried to set count_ as a static member, but boost doesn't allow me to do that either.

Comment (Laurawly) [reply gunrock/app/sm/sm_test.cuh]: I have a script for you if you wanna try it out: https://gist.github.com/Laurawly/7da2bfc2e0610705c7d3db3c57229c11

Comment (porumbes) [reply gunrock/app/sm/sm_test.cuh]: Ah, I see. You should be able to get rid of the `count_ptr` stuff if you use `std::ref` like so `  vf2_subgraph_iso(graph1, graph2, std::ref(callback));`

Comment (Laurawly) [reply gunrock/app/sm/sm_test.cuh]: yeah I can do ref as well.

File: gunrock/app/sm/sm_app.cuh [hunk 2]
```diff
@@ -229,16 +229,21 @@ double sm(
     bool quiet = parameters.Get<bool>("quiet");
     GraphT data_graph;
     GraphT query_graph;
+
+    gunrock::util::Location target = gunrock::util::HOST;
     // Assign pointers into gunrock graph format
     data_graph.CsrT::Allocate(num_nodes, num_edges, gunrock::util::HOST);
```
Comment (porumbes) [inline gunrock/app/sm/sm_app.cuh]: Let's make this allocation on `target` too.

File: gunrock/app/sm/sm_test.cuh [hunk 3]
```diff
@@ -156,7 +154,7 @@ double CPU_Reference(util::Parameters &parameters, GraphT &data_graph,
 
       // Print out all mappings
```
Comment (porumbes) [inline gunrock/app/sm/sm_test.cuh]: Please fix this comment. I don't think `callback` is  printing anything out here.

**Pull request merged**